ddtrace/contrib/trace_utils.py

```diff
@@ -415,3 +416,33 @@ def set_flattened_tags(
     for prefix, value in items:
         for tag, v in _flatten(value, sep, prefix, exclude_policy):
             span.set_tag(tag, processor(v) if processor is not None else v)
+
+
+def set_user(tracer, user_id, name="", email="", scope="", role="", session_id=""):
+    # type: (Tracer, str, Optional[str], Optional[str], Optional[str], Optional[str], Optional[str]) -> None
+    """Set user tags.
+    https://docs.datadoghq.com/logs/log_configuration/attributes_naming_convention/#user-related-attributes
+    https://docs.datadoghq.com/security_platform/application_security/setup_and_configure/?tab=set_tag&code-lang=python
+    """
+    span = tracer.current_root_span()
+    if span:
+        # Required unique identifier of the user
+        span.set_tag(user.ID, user_id)
+
+        # All other fields are optional
+        if name:
+            span.set_tag(user.NAME, name)
+        if email:
+            span.set_tag(user.EMAIL, email)
+        if scope:
+            span.set_tag(user.SCOPE, scope)
+        if role:
+            span.set_tag(user.ROLE, role)
+        if session_id:
+            span.set_tag(user.SESSION_ID, session_id)
+    else:
+        log.debug(
```

Comment:
should we raise an exception in this case since it is unexpected usage?

Reply:
I think an exception could be so intrusive: The users add this this function proactively into their code

Reply:
we should at least do a warning log

I worry that if someone thinks they are collecting user information and then nothing happens they won't know unless they enable debug logging


it might be be intuitive that you need to set the user during the lifetime of an active span.